sample-app/src/components/SearchBar.tsx

```diff
@@ -6,28 +6,38 @@ import '../sass/SearchBar.scss';
 import '../sass/Autocomplete.scss';
 import LoadingIndicator from './LoadingIndicator';
 import { useAutocomplete } from '../hooks/useAutocomplete';
+import { useRef } from 'react';
+import { AutocompleteResponse } from '@yext/answers-headless';
+import { executeSearchWithIntents } from '../utils/search-operations';
 
 const SCREENREADER_INSTRUCTIONS = 'When autocomplete results are available, use up and down arrows to review and enter to select.'
 
 interface Props {
   placeholder?: string,
   isVertical: boolean,
+  geolocationOptions?: PositionOptions,
   screenReaderInstructionsId: string
 }
 
 /**
  * Renders a SearchBar that is hooked up with an Autocomplete component
  */
-export default function SearchBar({ placeholder, isVertical, screenReaderInstructionsId }: Props) {
+export default function SearchBar({
+  placeholder,
+  isVertical,
+  geolocationOptions,
+  screenReaderInstructionsId 
+}: Props) {
   const answersActions = useAnswersActions();
   const query = useAnswersState(state => state.query.input);
-  const [ autocompleteResults, executeAutocomplete ] = useAutocomplete(isVertical);
   const isLoading = useAnswersState(state => state.searchStatus.isLoading);
+  const responseToLatestRequestRef = useRef<Promise<AutocompleteResponse|undefined>>();
+  const [ autocompleteResponse, executeAutocomplete] = useAutocomplete(isVertical);
 
-  function executeQuery () {
-    isVertical
-      ? answersActions.executeVerticalQuery()
-      : answersActions.executeUniversalQuery();
+  async function executeQuery () {
+    const responseToLatestRequest = await responseToLatestRequestRef.current;
```

Comment:
Would `latestAutocompleteResponse` be clearer?

Reply:
hmm it's not exactly the latest autocomplete response, but rather the promise stored in this ref is tied to the latest request. Let's say there's an autocomplete request #1 trigger before and an autocomplete request #2 trigger after a user hit 'enter' to execute search, and response #2 comes in. #2 would be the latest autocomplete response, but the await here is waiting on the response to the latest request right before the search execution.

Reply:
got it. I was also thinking that it would be helpful to include 'Autcomplete' in the name somewhere, however I'll leave that up to you if we should include that because that would result in a really long name

Reply:
I think it's ok to keep it as is for now? If there will be more responses to manage in searchBar in the future then we can make that differentiation more distinct in the names